src/impression.js

```diff
@@ -156,15 +165,25 @@ function applyResponse(win, viewer, response) {
     new Image().src = trackUrl;
   }
 
-  // Replace the location href params with new location params we get.
+  // Replace the location href params with new location params we get (if any).
   if (adLocation) {
-    if (!win.history.replaceState) {
-      return;
-    }
-    const currentHref = win.location.href;
-    const url = parseUrl(adLocation);
-    const params = parseQueryString(url.search);
-    const newHref = addParamsToUrl(currentHref, params);
-    win.history.replaceState(null, '', newHref);
+    replaceUrl(win, adLocation);
   }
 }
+
+/**
+ * Function to replace the location href params
+ * @param {!Window} win
+ * @param {string} replaceUrl
+ */
+function replaceUrl(win, replaceUrl) {
+  if (!win.history.replaceState) {
+    return;
+  }
+
+  const currentHref = win.location.href;
+  const url = parseUrl(replaceUrl);
+  const params = parseQueryString(url.search);
+  const newHref = addParamsToUrl(currentHref, params);
```

Comment:
Do we only care about the params of the replaceUrl? (I.e., the other parts [path, origin] we ignore?) Just want to double check.

Reply:
Origin should always be the same. I believe path should be the same as well. ping @sucharithav @cramforce  to confirm : )

Reply:
hmmm. @sucharithav when you say 
> Path could be different as well

You mean origin and path can both be different? I am thinking of replacing the url as what we do with initial `replaceUrl` [here](https://github.com/ampproject/amphtml/blob/master/src/service/viewer-impl.js#L385). Where we request origin to be the same. And then replace everything other than location.hash

Questions are: 
1. Should I check for same origin?
2. Does it make sense to replace everything other than hash?

Reply:
Reusing the existing replaceUrl code will work - origin will be the same since the AMP Viewer is going to be generating the replaceURL relative to the current AMP runtime URL in the first place. Safe to confirm origins match, but as long as the viewer's doing the right thing, origins should always match.